src/Agent/NewRelic/Agent/Core/Config/ConfigurationLoader.cs

```diff
@@ -484,7 +484,13 @@ public string GetFullLogFileName()
 
         private string GetLogFileName()
         {
-            string name = fileName;
+            string name = System.Environment.GetEnvironmentVariable("NEW_RELIC_LOG");
+            if (name != null)
+            {
+                return name;
+            }
+
+            name = fileName;
```

Comment:
The first if (name != null) block if true will never hit the following if block (name != null) especially since the conditions are similar.  Is it true that you never want to hit the following if block that returns return Strings.SafeFileName(name)?

Reply:
yes. if the NEW_RELIC_LOG env var is set, that takes precedence over any other naming.